### ISSUE TITLE
Add ItemStack sensitive version of Item.getFoodProperties(), called from FoodStats.eat(...)

### ISSUE THREAD

**Issue by queekusme** [MinecraftForge/MinecraftForge#7701]
When eating an item `PlayerEntity.eat(World, ItemStack)` calls `this.getFoodData().eat(...)`, which takes the `Item` and `ItemStack` versions of the same food item.

`FoodStats.eat(Item, ItemStack)` gets the food properties by calling `Food food = item.getFoodProperties();`

This means that only one set of food stats can be defined for an item type.

It would be desirable if `Item.getFoodProperties` could take ItemStack and have the ItemStack passed into it so that information such as NBT data can be taken into account

e.g.

```
@Nullable
public Food getFoodProperties(ItemStack stack)
{
    return this.foodProperties;
}
```
Called as

```
// In FoodStats' method, eat(...)
Food food = item.getFoodProperties(itemStack);
```

**Comment by TheCurle** [comment]
Dear god, now there are three of them..
Closed by #7702 and #7543

**Comment by queekusme** [comment]
Closing as this has been implemented in another PR
